ccmlib/node.py

```diff
@@ -745,30 +745,59 @@ def stop(self, wait=True, wait_other_notice=False, other_nodes=None, gently=True
         else:
             return False
 
-    def wait_for_compactions(self, idle_timeout=300):
+    def _get_pending_tasks(self, stats, keyspace, column_family):
+        assert keyspace is not None, 'keyspace is not specified'
+        assert column_family is not None, 'column_family is not specified'
+        # "nodetool compactionstats" prints the compaction stats like:
+        # pending tasks: 42
+        # - ks1.cf1: 13
+        # - ks1.cf2: 19
+        # - ks2.cf1: 10
+        pattern = re.compile(r'\-\s+([^:]+):\s*(\d+)')
```

Comment:
why not just use the regex to parse ks and cf for us to save the string format below?

Reply:
Also, specifying just the keyspace seems like a valid use case, so we can sum all the compaction jobs from all lines matching the given keyspace.

Reply:
> why not just use the regex to parse ks and cf for us to save the string format below?

this is the first version i had. but i changed to existing one. because the more regex the more convoluted. and why is it better than what we have? we are concatenating the ks and cf everywhere.

Reply:
> Also, specifying just the keyspace seems like a valid use case, so we can sum all the compaction jobs from all lines matching the given keyspace.

can do. but i don't want to over-engineering. the existing implementation is able to address the need, and it is extensible.

Reply:
It would be different if it was a utility function in dtest itself.  But since it is a library it should be extended in a more general way for future use cases, even if there's no immediate need.
I don't think the extension is that difficult and you're already touching the code it can be puggybacked on that relatively easily.

Reply:
i want to invest my time in something we really need. if we have a real world use case, i would be happy to implement this. can we implement this feature when we need it?

Reply:
done.